### ISSUE TITLE
[manual] Add client side search

### ISSUE THREAD

**Issue by oschuett** [cp2k/cp2k#343]
The manual currently uses a Google Custom Search. Unfortunately, this gives us only limited control over which pages get indexed. Out of the 19925 listed pages google chooses to indexes only 4348. This isn't really surprising given how little traffic the site receives. And with every release we are adding 2218 more pages. Hence, we should at some point move to a client-side search engine like [Lunr.js](https://lunrjs.com).

**Comment by ltalirz** [comment]
Just to mention some corollary of this 

![image](https://user-images.githubusercontent.com/1053245/107684458-1bfd8280-6ca3-11eb-86a3-3d96bc049956.png)

**Comment by oschuett** [comment]
Yeah, it's bad. Unfortunately, I won't have the bandwidth for it (this year).
So, contributions are always welcome ;-)

**Comment by mattatlincoln** [comment]
Can we just add site:manual.cp2k.org onto the search string? Presumably it is not that simples.


[University of Lincoln]<http://lincoln.ac.uk/>  Dr. Matt Watkins | Associate Professor
School of Mathematics and Physics
University of Lincoln. Brayford Pool, Lincoln, Lincolnshire. LN6 7TS
tel: +44 (0)1522 835867
mattatlincoln.github.io<https://mattatlincoln.github.io/>


________________________________

                       
                        
The University of Lincoln, UK, is the Modern University of the Year in The Times and Sunday Times Good University Guide 2021. We are TEF-Gold rated and recognised for our high student satisfaction, award-winning industry links and world-leading research. Visit www.lincoln.ac.uk.  
                         
The information in this e-mail and any attachments may be confidential. If you have received this email in error please notify the sender immediately and remove it from your system. Do not disclose the contents to another person or take copies.
​
Email is not secure and may contain viruses. The University of Lincoln makes every effort to ensure email is sent without viruses, but cannot guarantee this and recommends recipients take appropriate precautions.
​
The University may monitor email traffic data and content in accordance with its policies and English law. Further information can be found at: http://www.lincoln.ac.uk/legal.

**Comment by oschuett** [comment]
> Can we just add site:manual.cp2k.org onto the search string?

That's essentially what we are already doing with the [custom search box](https://programmablesearchengine.google.com).
The problem is that Google does not index, ie. crawl, all our pages because we have too many.

**Comment by ltalirz** [comment]
Maybe Algolia DocSearch is worth a try? https://docsearch.algolia.com/

Free for open-source projects, just read the [checklist](https://docsearch.algolia.com/docs/who-can-apply/) and [apply to the program](https://docsearch.algolia.com/apply).

It will crawl once a week

**Comment by oschuett** [comment]
Great find! I've just applied🤞

**Comment by oschuett** [comment]
Good news: We're getting sponsored by Algolia DocSearch :-)

Bad news: Their [crawler](https://www.algolia.com/doc/tools/crawler/guides/extracting-data/) expects [semantic HTML](https://en.wikipedia.org/wiki/Semantic_HTML) while our manual pages are build with nested tables... which was the style at the time.

So, I think we first need a new manual generator. While we're at it we could switch to [Sphinx with Markdown](https://www.sphinx-doc.org/en/master/usage/markdown.html).

**Comment by oschuett** [comment]
The new Sphinx-based manual has a builtin search. While it's not perfect, I think it's good enough. One could probably tweak it a bit more via the [Sphinx Config](https://github.com/cp2k/cp2k/blob/master/docs/conf.py).

So, if nobody objects, I will close our Algolia account again.

**Comment by ltalirz** [comment]
Just would suggest to test the quality of search results before you cancel the account; it can make a real difference for users of the documentation

**Comment by he-zilong** [comment]
Is this issue still relevant now that the Sphinx-readthedocs manual has been working for a couple of years with the native search box?

**Comment by oschuett** [comment]
Yes, we can close this issue. The sphinx native search is far from perfect, but we won't be going back to a server-side solution. If anything, we should open a new issue for improving the native search.

**Comment by oschuett** [comment]
For posterity, we've now adopted Pagefind: #6047.
